app/models/user.rb

```diff
@@ -427,4 +428,8 @@ def validate_username_length
         maximum: User.username_max_length)
       validator.validate(self)
     end
+
+    def add_unsubscribe_hash
+      self.unsubscribe_hash = SecureRandom.hex
```

Comment:
While this is secure, hexadecimal strings look like MD5 hashes and don't feel so secure :thinking:. So, just in order to make the string look more random, how about:

```suggestion
      self.unsubscribe_hash = SecureRandom.base58(32) 
```

I think we can't use `SecureRandom.base64` in this case because it might contain a slash, which is a special character in a URL.